src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Integer.Unsigned.D.cs

```diff
@@ -0,0 +1,309 @@
+            uint firstDigit = text[0] - 48u; // '0'
+            if (firstDigit > 9)
+            {
+                bytesConsumed = 0;
+                value = default;
+                return false;
+            }
+            uint parsedValue = firstDigit;
+
+            if (text.Length < ParserHelpers.Int16OverflowLength)
+            {
+                // Length is less than Parsers.Int16OverflowLength; overflow is not possible
+                for (int index = 1; index < text.Length; index++)
+                {
+                    uint nextDigit = text[index] - 48u; // '0'
+                    if (nextDigit > 9)
+                    {
+                        bytesConsumed = index;
+                        value = (ushort)(parsedValue);
+                        return true;
+                    }
+                    parsedValue = parsedValue * 10 + nextDigit;
+                }
+            }
+            else
+            {
+                // Length is greater than Parsers.Int16OverflowLength; overflow is only possible after Parsers.Int16OverflowLength
+                // digits. There may be no overflow after Parsers.Int16OverflowLength if there are leading zeroes.
+                for (int index = 1; index < ParserHelpers.Int16OverflowLength - 1; index++)
+                {
+                    uint nextDigit = text[index] - 48u; // '0'
+                    if (nextDigit > 9)
+                    {
+                        bytesConsumed = index;
+                        value = (ushort)(parsedValue);
+                        return true;
+                    }
+                    parsedValue = parsedValue * 10 + nextDigit;
+                }
+                for (int index = ParserHelpers.Int16OverflowLength - 1; index < text.Length; index++)
+                {
+                    uint nextDigit = text[index] - 48u; // '0'
+                    if (nextDigit > 9)
+                    {
+                        bytesConsumed = index;
+                        value = (ushort)(parsedValue);
+                        return true;
+                    }
+                    // If parsedValue > (ushort.MaxValue / 10), any more appended digits will cause overflow.
+                    // if parsedValue == (ushort.MaxValue / 10), any nextDigit greater than 5 implies overflow.
+                    if (parsedValue > ushort.MaxValue / 10 || (parsedValue == ushort.MaxValue / 10 && nextDigit > 5))
+                    {
+                        bytesConsumed = 0;
+                        value = default;
+                        return false;
+                    }
+                    parsedValue = parsedValue * 10 + nextDigit;
+                }
+            }
+
+            bytesConsumed = text.Length;
+            value = (ushort)(parsedValue);
+            return true;
+        }
+
+        private static bool TryParseUInt32D(ReadOnlySpan<byte> text, out uint value, out int bytesConsumed)
```

Comment:
Consider applying the same loop unrolling optimization as TryParseInt32D (https://github.com/dotnet/corefx/pull/25078/files#diff-5e90497df430f6541664f0a780acac46R199).

Same with TryParseUInt16D and TryParseByteD.

Reply:
Not in this PR. We need to get this in before I leave for vacation and this sort of thing is too much code churn for a commit this size.

Reply:
Agreed :)
Already pushing the bar for what can be reviewed within the browser.